### ISSUE TITLE
how the model was fine tuned?

### ISSUE THREAD

**Issue by betterze** [justinpinkney/stable-diffusion#5]
Dear Justin,

Thank you for sharing this great model. I really enjoy it. 

Could you tell me how you fine tune this model? You fine tune both the CLIP image encoder and the diffusion unet? or you fix the weight of diffusion unet? 

Thank you for your help.

Best Wishes,

Zongze

**Comment by fugokidi** [comment]
@betterze I think it was fine-tuned only the diffusion model if I am not mistaken.

**Comment by justinpinkney** [comment]
Yes, the image encoder is frozen and the unet is fine tuned, you can see the training config here: https://github.com/justinpinkney/stable-diffusion/blob/main/configs/stable-diffusion/sd-image-condition-finetune.yaml

**Comment by leod** [comment]
Hi @justinpinkney, thank you for sharing this amazing work! The linked training config uses a learning rate of 1e-4 with a batch size of 6. Is this a good learning rate to use for fine tuning the UNet? I'm wondering if it is a bit large for such a big model.

**Comment by justinpinkney** [comment]
That's right, although batch size was 24 as I used 4 GPUs. 

I didn't try tuning any hyper parameters, so have no reason to think that's the best learning rate. But saying that I didn't see any instability, probably the small domain gap and Adam makes it quite forgiving.

**Comment by ozanciga** [comment]
> Yes, the image encoder is frozen and the unet is fine tuned, you can see the training config here: https://github.com/justinpinkney/stable-diffusion/blob/main/configs/stable-diffusion/sd-image-condition-finetune.yaml

hey @justinpinkney is the autoencoder frozen as well? seems like not from the config file. if not, why? asking because i found the autoencoder is pretty capable of encoding images regardless of the style (cartoon etc) or resolution (works for 256^2 to 1024^2 images). i would assume if the reconstruction is reasonable, then we wouldn't need to train that part.

**Comment by pfdamasceno** [comment]
Hi @justinpinkney, this is fun. Thanks for putting the time!

Is there a way to know when the model has converged, other than looking at the validation images? 

Training for 1000 epochs (400,000 steps, BATCH_SIZE = 1, N_GPUS = 2, ACCUMULATE_BATCHES = 4) still produces simple pokemons for any prompt (Obama / Trump etc) instead of your president ones. Am I missing something? Thanks!
